yarn-project/telemetry-client/src/otel.ts

```diff
@@ -0,0 +1,53 @@
+import { Meter } from '@opentelemetry/api';
+import { OTLPMetricExporter } from '@opentelemetry/exporter-metrics-otlp-http';
+import { HostMetrics } from '@opentelemetry/host-metrics';
+import { Resource } from '@opentelemetry/resources';
+import { MeterProvider, PeriodicExportingMetricReader } from '@opentelemetry/sdk-metrics';
+import { SEMRESATTRS_SERVICE_NAME, SEMRESATTRS_SERVICE_VERSION } from '@opentelemetry/semantic-conventions';
+
+import { TelemetryClient } from './telemetry.js';
+
+export class OpenTelemetryClient implements TelemetryClient {
+  hostMetrics: HostMetrics | undefined;
+  protected constructor(private resource: Resource, private meterProvider: MeterProvider) {}
+
+  getMeter(name: string): Meter {
+    return this.meterProvider.getMeter(name, this.resource.attributes[SEMRESATTRS_SERVICE_VERSION] as string);
+  }
+
+  public start() {
+    this.hostMetrics = new HostMetrics({
+      name: this.resource.attributes[SEMRESATTRS_SERVICE_NAME] as string,
+      meterProvider: this.meterProvider,
+    });
+
+    this.hostMetrics.start();
+  }
+
+  public async stop() {
+    await Promise.all([this.meterProvider.shutdown()]);
```

Comment:
Does this also shut down the hostMetrics?

Reply:
The host metrics does not expose a `stop` method :(